csharp-advanced-topics/ActionFuncDelegates/ActionAndFuncDelegatesTest/ActionFuncDelegates_Tests.cs

```diff
@@ -0,0 +1,45 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NUnit.Framework.Internal;
+using ActionAndFuncDelegates;
+using System.Collections.Generic;
+using System;
+using System.Linq;
+
+namespace ActionAndFuncDelegatesTest
+{
+    [TestClass]
+    public class ActionFuncDelegates_Tests
+    {
+        [TestMethod]
+        public void TestFunc()
+        {
+            List<int> numbers = new List<int> { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
+            Func<int, bool> isEvenFunc = x => x % 2 == 0;
+
+            var evenNumbersByAnonymousMethod = numbers.Where(x => x % 2 == 0); //Use a lambda expression
+            var evenNumbersByFunc = numbers.Where(isEvenFunc); //Use a Func
+            var evenNumbersByExistingMethod = numbers.Where(ActionsAndFuncs.IsNumberEven); //Use a previously declared method
+
+            Assert.AreEqual(5, evenNumbersByAnonymousMethod.Count());
+            Assert.AreEqual(5, evenNumbersByFunc.Count());
+            Assert.AreEqual(5, evenNumbersByExistingMethod.Count());
+
+        }
+
+        [TestMethod]
+        public void TestAction()
```

Comment:
Please use the naming convention from the guidelines. It's strange at the beginning, but you'll get used to it.